src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskScheduler.cs

```diff
@@ -353,7 +353,7 @@ public int Id
         {
             get
             {
-                if (m_taskSchedulerId == 0)
+                if (Volatile.Read(ref m_taskSchedulerId) == 0)
```

Comment:
TaskScheduler.Id is rarely used on hot paths; I don't think it's worth the extra code complication with Volatile.Read to avoid an extra fence on arm. If we really want to do something here,  though, I'd prefer it be structured as:
```C#
int id = m_taskSchedulerId;
if (id == 0)
{
    ... // existing lazy init logic
    id = m_taskSchedulerId;
}
return id;
```
leaving the field volatile.